### PR TITLE
(PXP-6184): refactor/nodeCountTitle: make nodeCountTitle field optional

### PR DESCRIPTION
**Pull request by ocshawn**
Jira Ticket: [PXP-6184](https://ctds-planx.atlassian.net/browse/PXP-6184)

### Improvements
make the nodeCountTitle field optional by using a reformatted dataType if not filled out
Adds a labelToTitle helper function for styling dataType as well as tests for this new function

### CODE REVIEW

File: docs/multi_tab_explorer.md [hunk 1]
```diff
@@ -152,7 +152,7 @@ An example of this new `explorerConfig` is (some contents are omitted for concis
 - `filters`: Required.
 - `guppyConfig`: Required.
   - `dataType`: Required. Must match the index “type” in the guppy configuration block in the manifest.json.
-  - `nodeCountTitle`: Required. Plural of root node.
+  - `nodeCountTitle`: Optional. Plural of root node.
```
Comment (mfshao) [inline docs/multi_tab_explorer.md:155]: can you add a bit of text here explaining what will happen if this is omitted?

**Review by mfshao — COMMENTED** [review]
generally lgtm, one small comment

File: src/GuppyDataExplorer/utils.js [hunk 2]
```diff
@@ -57,3 +59,17 @@ export const humanizeNumber = (number, fixedPoint = 2) => {
...
   return Number.parseFloat(number).toExponential(fixedPoint);
 };
+
+/*
+* Convert label to pluralized Title
+* @param {label} string - a label to convert to title
+* @param {titleCase} boolean - Should first letter be capitalized
+* @returns {string} Pluralized formatted word
+*/
+export const labelToTitle = (label, titleCase = true) => {
```
Comment (paulineribeyre) [inline src/GuppyDataExplorer/utils.js:69]: i would rename it "labelToPlural" or "labelToPluralTitle"

**Review by mfshao — APPROVED** [review]
thanks 👍

File: docs/portal_config.md [hunk 3]
```diff
@@ -265,7 +265,7 @@ Below is an example, with inline comments describing what each JSON block config
     ],
     "guppyConfig": { // required; how to configure Guppy to work with the Data Explorer
       "dataType": "case", // required; must match the index “type” in the guppy configuration block in the manifest.json
-      "nodeCountTitle": "Cases", // required; plural of root node
+      "nodeCountTitle": "Cases", // optional; plural of root node
```
Comment (paulineribeyre) [inline docs/portal_config.md:268]: should update the description here too

File: docs/portal_config.md [hunk 4]
```diff
@@ -319,7 +319,7 @@ Below is an example, with inline comments describing what each JSON block config
       "fieldMapping": [ // optional; a way to rename GraphQL fields to be more human readable
         { "field": "object_id", "name": "GUID" } // required; the file index should always include this one
       ],
-      "nodeCountTitle": "Files", // required; plural of root node
+      "nodeCountTitle": "Files", // optional; plural of root node
```
Comment (paulineribeyre) [inline docs/portal_config.md:322]: and here too

**Pull request merged**